File: frontend/src/app/modules/menu/components/ticket/ticket.ts
export class Ticket{
  idTicket: string;
  name: string;
  cost: string;
  visits: string;
  duration: string;

  static cloneBase(ticket: Ticket): Ticket {
    const clonedTicket: Ticket = new Ticket();
    clonedTicket.idTicket = ticket.idTicket;
    clonedTicket.name = ticket.name;
    clonedTicket.cost = ticket.cost;
    clonedTicket.visits = ticket.visits;
    clonedTicket.duration = ticket.duration;
    return clonedTicket;
  }
}

File: fapi/src/main/java/com/gym/fapi/models/Ticket.java
package com.gym.fapi.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Ticket {
    private int idTicket;
    private int cost;
    private String duration;
    private String visits;
    private String name;

    public Ticket() {
    }

    public Ticket(int idTicket, int cost, String duration, String visits, String name) {
        this.idTicket = idTicket;
        this.cost = cost;
        this.duration = duration;
        this.visits = visits;
        this.name = name;
    }

    public int getIdTicket() {
        return idTicket;
    }

    public void setIdTicket(int idTicket) {
        this.idTicket = idTicket;
    }

    public int getCost() {
        return cost;
    }

    public void setCost(int cost) {
        this.cost = cost;
    }

    public String getDuration() {
        return duration;
    }

    public void setDuration(String duration) {
        this.duration = duration;
    }

    public String getVisits() {
        return visits;
    }

    public void setVisits(String visits) {
        this.visits = visits;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}
